darts/models/forecasting_model.py

```diff
@@ -251,13 +250,26 @@ def historical_forecasts(self,
                 train_cov = covariates.drop_after(pred_time)
 
             if retrain:
-                if covariates and 'covariates' in fit_signature.parameters:
+                if covariates and "covariates" in fit_signature.parameters:
                     self.fit(series=train, covariates=train_cov)
+                elif covariates and "exog" in fit_signature.parameters:
+                    self.fit(series=train, exog=train_cov)
                 else:
                     self.fit(series=train)
 
-            if covariates and 'covariates' in predict_signature.parameters:
-                forecast = self.predict(n=forecast_horizon, series=train, covariates=train_cov)
+            if covariates:
+                if "covariates" in predict_signature.parameters:
+                    if 'series' in predict_signature.parameters:
+                        forecast = self.predict(n=forecast_horizon, series=train, covariates=train_cov)
+                    else:
+                        forecast = self.predict(n=forecast_horizon, covariates=train_cov)
+                else:
+                    if 'series' in predict_signature.parameters:
+                        forecast = self.predict(n=forecast_horizon, series=train, exog=train_cov)
+                    else:
+                        start = train.end_time() + train.freq()
+                        train_cov = covariates[start:start+(forecast_horizon-1)*train.freq()]
+                        forecast = self.predict(n=forecast_horizon, exog=train_cov)
```

Comment:
I guess this could fail if the method is called with `covariates` argument, but the model supports `exog`?

Reply:
I refactored the code here a bit. I'm not sure if it's better at all. It's a little inconvenient that currently the RegressionModel is as far as I know the only model that checks if the exogenous variables are starting at time `last_train_timestamp+1`. That's why it needs to be treated separately.
It needs to check that because it reuses the last few values from training for prediction. For example, if you currently enter 12 values for the exog variable you get 12 values back. Maybe it would be better that if you use a lag of 12 and pass 12 values you only get 1 value back. The data is used to construct the lagged dataframe which in turn is used to make one prediction. 
I'll take a look how this is handled elsewhere to be consistent here.